### DISCUSSION TITLE
Should gltf-transform validate indicate error status via return code?

### DISCUSSION THREAD

**Discussion by zeux** [donmccurdy/glTF-Transform#1420] [Ideas]
`gltf-transform validate` appears to return 0 as error code even if the input asset contains errors. Would it make sense to treat presence of validation errors as an error of the command as well? I expected this because that's the behavior of official glTF Validator binary so I got used to it in shell scripts.

**Comment by donmccurdy** [comment]
Yes — my feeling is that the CLI should return a non-zero status code if the validation process identifies errors. The dependency currently handling the CLI interface doesn't appear to provide a good way to do this, but I'll track the goal in a new issue.

Related:

- https://github.com/donmccurdy/glTF-Transform/issues/1425
- https://github.com/donmccurdy/glTF-Transform/issues/1251#issuecomment-1953344244
